### PR TITLE
Localise parts of the beatmap listing overlay

### PR DESCRIPTION
**Pull request by smoogipoo**
As discussed in today's meeting, just an example consuming osu!web localisations.

We'll need to add an attribute to support localisable enum names (`[LocalisableDescription]`?)

### CODE REVIEW

File: osu.Game/Overlays/BeatmapListing/BeatmapSearchScoreFilterRow.cs
Comment (peppy) [inline osu.Game/Overlays/BeatmapListing/BeatmapSearchScoreFilterRow.cs:65]: Bit wary of throwing in a function which could cause a crash of the game from an incorrect external (api) value.

**Review by peppy — APPROVED** [review]
Seems fine apart from pointed out issue, will wait for feedback before merging.

File: osu.Game/Overlays/BeatmapListing/BeatmapSearchScoreFilterRow.cs
Comment (bdach) [reply osu.Game/Overlays/BeatmapListing/BeatmapSearchScoreFilterRow.cs:65]: It's not actually an external API value in this case. These are being instantiated via

https://github.com/ppy/osu/blob/7654df94f6f37b8382be7dfcb4f674e03bd35427/osu.Game/Overlays/BeatmapListing/BeatmapSearchFilterRow.cs#L83-L84

i.e. all items here should already be defined in the enum. I'd say it's *probably* fine to crash on that, but for ranks `ToString()` may also be an ok default.

Comment (peppy) [reply osu.Game/Overlays/BeatmapListing/BeatmapSearchScoreFilterRow.cs:65]: Right. Just thinking as a general rule though, we probably want to stop throwing on these kinds of lookups as it's more likely to cause issues than help.

Comment (smoogipoo) [reply osu.Game/Overlays/BeatmapListing/BeatmapSearchScoreFilterRow.cs:65]: I would rather things throw than silently fail? If we ever add an SSSH or something like that.

Comment (peppy) [reply osu.Game/Overlays/BeatmapListing/BeatmapSearchScoreFilterRow.cs:65]: Well consider the case we forget to open beatmap listing and it somehow doesn't throw in a test and makes it into a release: we either have users reporting "why is SSSH showing as 19?" or "OH MY GOD BEATMAP LISTING IS CRASHING THE GAME ON OPENING".

food for thought anyways, i don't have any strong opinion for now.

**Comment by bdach** [comment]
As a semi-related aside, the [language enum](https://github.com/ppy/osu/blob/master/osu.Game/Localisation/Language.cs) still doesn't have any other values than English/Japanese, so the other languages aren't really consumable yet. I can add them in a separate PR if you don't have that queued already.

**Comment by smoogipoo** [comment]
Nope, go ahead and add. The only reason I didn't was because I tried to add a language and it required unicode, which needs a different font and relevant dropdown implementation.

**Comment by peppy** [comment]
As a temporary measure the non-unicode alternative could be used for said languages with an inline TODO for future reference.

**Comment by smoogipoo** [comment]
I'll leave it up to @bdach in any case :)

**Comment by bdach** [comment]
Yeah there's a fair bit of work involved in making Arabic script work (add font, add LTR text support)... I think I'll just skip the languages that can't work yet for now, leave a TODO for those and just add what I can.

**Pull request merged**
